File: 卡码/代码随想录/C++/数组/02-双指针/doublePointer.cpp
//27. 移除元素
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
	int removeElement(vector<int>& nums, int val) {
		int slow=0;
		for(int fast=0;fast<nums.size();fast++){
			if(nums[fast]!=val) nums[slow++]=nums[fast];
		}
		return slow;
	}
};

int main(){
	vector<int> nums={0,1,2,2,3,0,4,2};
	Solution solution=new Solution();
	printf("%d\n",solution->removeElement(nums,2));
	return 0;
}

File: 卡码/代码随想录/1/01-数组/Java/src/_02_双指针/DoublePointer.java
//27. ÒÆ³ıÔªËØ
package _02_Ë«Ö¸Õë;

public class DoublePointer {
	public static void main(String[] args) {
		int[] nums=new int[] {0,1,2,2,3,0,4,2};
		Solution solution = new Solution();
		System.out.println(solution.removeElement(nums, 2));
	}
}
class Solution {
  public int removeElement(int[] nums, int val) {
  	int slow=0,fast=0;
  	for(fast=0;fast<nums.length;fast++) {
  		if(nums[fast]!=val) nums[slow++]=nums[fast];
  	}
  	return slow;
  }
}